[Automation] Fix AttributeError when overwrite_registry is None

alxtkr77 opened the pull request:
## Problem
The `patch_remote.py` script crashes with AttributeError when `overwrite_registry` is None.

**Reference**: ML-11462

## Solution
Add None check before calling `.endswith()` on `overwrite_registry` parameter.

## Changes
- `automation/patch_igz/patch_remote.py:636` - Added None check: `if overwrite_registry and overwrite_registry.endswith("/")`

Commit:
[Automation] Fix AttributeError when overwrite_registry is None

Reference: ML-11462

liranbg commented on automation/patch_igz/patch_remote.py:
```diff
@@ -633,7 +633,7 @@ def _resolve_overwrite_registry(self):
             docker_registry = docker_registry[:-1]
-        if overwrite_registry.endswith("/"):
+        if overwrite_registry and overwrite_registry.endswith("/"):
             overwrite_registry = overwrite_registry[:-1]
```
```suggestion
        if docker_registry:
            docker_registry = docker_registry.rstrip("/")
        if overwrite_registry:
            overwrite_registry = overwrite_registry.rstrip("/")
```

alxtkr77 replied:
✅ Applied! Changed to use `rstrip("/")` for both registries. Much cleaner - thanks for the suggestion! 

Also added None check for `docker_registry` for consistency.

Commit:
[Automation] Use rstrip() for cleaner trailing slash removal

Address review feedback from liranbg to use rstrip('/') instead of
endswith() check for removing trailing slashes from registry URLs.

Benefits:
- More Pythonic and concise
- Handles multiple trailing slashes
- Cleaner code

Reference: ML-11462
